skypalwarmup/JSUserRegistrationService/src/main/java/application/service/UserServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package application.service;
+
+import application.dao.UserRepository;
+import application.dto.UserInputDTO;
+import application.entity.User;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/**
+ * Copyright (c) 2016-2018, Jumia.
+ */
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    private static final Log logger = LogFactory.getLog(UserServiceImpl.class);
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Override
+    public Boolean register(UserInputDTO userInputDTO) {
```

Comment:
Line 25: This service belongs to our domain logic. Hence, it should already receive a `User` and not a `UserInputDTO`. The reason is that we don't want to couple payload changes (DTO) to our domain. Hence, the `mapFromUserDTO` should be done before the service called. E.g., the `Controller` might call a `Mapper` class that receives a `UserInputDTO` and returns a `User`. The `User` would be then passed to the `register` method.

Alternatively, we could create a `Facade` that would receive a `UserInputDTO`, maps it and, internally, calls the service for user registration. What do you think?

To sum up: Avoid passing DTO to our domain model.

Reply:
I think a `Facade` is an overkill in this case, since we only have 1 service, and one type of DTO which maps to a `User`. I like the first approach, and it makes a lot of sense. I was thinking about making domain models for the service layer separate from the Entities which could be very specific to the persistence layer, but for the sake of the warmup, it's not needed.